lib/matplotlib/cbook/__init__.py

```diff
@@ -2033,6 +2033,20 @@ def _setattr_cm(obj, **kwargs):
     """
     sentinel = object()
     origs = [(attr, getattr(obj, attr, sentinel)) for attr in kwargs]
+
+    # we only want to set back things that are in the instance dict or
+    # are properties.  Everything else we are either introducing to begin with
+    # or are shadowing from something later in the resolution order.
+    def filter_restores(obj, attr, val):
+        if attr in obj.__dict__:
+            return True
+
+        if isinstance(getattr(type(obj), attr), property):
+            return True
+        return False
+
+    origs = [(attr, val if filter_restores(obj, attr, val) else sentinel)
+             for attr, val in origs]
```

Comment:
Should be able to squeeze these two loops together?
```suggestion

    # we only want to set back things that are in the instance dict or
    # are properties.  Everything else we are either introducing to begin with
    # or are shadowing from something later in the resolution order.
    def get_filtered_attr(obj, attr):
        val = getattr(obj, attr, sentinel)

        if attr in obj.__dict__:
            return val
        if isinstance(getattr(type(obj), attr), property):
            return val

        return sentinel

    origs = [(attr, get_filtered_attr(obj, attr)) for attr in kwargs]
```